### PR TITLE
Add flags to setup and run scripts

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add flags to setup and run scripts to avoid hidden failure (b/322868084).

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
Upload to Airflow and test.

**List links for your tests (use go/shortn-gen for any internal link):** ...
Successfully make the run_model task fail: [link](https://screenshot.googleplex.com/4cguJZbvcBuHP8D)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: xlml/apis/test_config.py [hunk 1]
```diff
@@ -204,11 +204,11 @@ def benchmark_id(self) -> str:
 
   @property
   def setup_script(self) -> Optional[str]:
-    return '\n'.join(self.set_up_cmds)
+    return '\n'.join(('set -xue', *self.set_up_cmds))
```
Comment (will-cromar) [inline xlml/apis/test_config.py:207]: What do you think of adding this to the SSH implementation instead? I'm thinking that if these run in a Docker container, they won't be guaranteed to run in `bash`. Within our SSH function, we can guarantee that

Comment (RissyRan) [reply xlml/apis/test_config.py:207]: I don't have a strong preference on this. I will expect a new Test like `TpuVmTest(TestConfig[Tpu])` will be added for docker container, and we should have control there when calling `setup_script` or `run_script`. For SSH connection, we need to add logic into GPU and TPU implementation. So both strategies will add changes to multiple places.

Comment (will-cromar) [reply xlml/apis/test_config.py:207]: Okay, let's try it this way. If needed, we can explicitly wrap the commands in `bash -cxue` to ensure the shell options are set.

**Review by will-cromar — APPROVED** [review]

File: xlml/apis/test_config.py (hunk 1 above)
Comment (RissyRan) [reply xlml/apis/test_config.py:207]: SGTM! Thanks!

**Pull request merged**